src/main/java/io/aiven/kafka/connect/http/sender/OAuth2HttpSender.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2021 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.connect.http.sender;
+
+import java.io.IOException;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.util.Map;
+
+import org.apache.kafka.connect.errors.ConnectException;
+
+import io.aiven.kafka.connect.http.config.HttpSinkConfig;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+final class OAuth2HttpSender extends HttpSender {
+
+    private final Logger logger = LoggerFactory.getLogger(OAuth2HttpSender.class);
+
+    private String accessTokenAuthHeader;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    OAuth2HttpSender(final HttpSinkConfig config) {
+        super(config);
+    }
+
+    //for testing
+    OAuth2HttpSender(final HttpSinkConfig config, final HttpClient httpClient) {
+        super(config, httpClient);
+    }
+
+    @Override
+    protected void sendWithRetries(final HttpRequest.Builder requestBuilder,
+                                   final HttpResponseHandler originHttpResponseHandler) {
+        final HttpResponseHandler handler = response -> {
+            if (response.statusCode() == 401) { // access denied or refresh of a token is needed
```

Comment:
Implementation is absolutely fine, but I would rather make some scheduled task to refresh tokens and do not interrupt a flow.

Reply:
I wonder if it's really worth to increase the complexity...